File: 1200. Minimum Absolute Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& a) {
        sort(a.begin(), a.end());
        int dis=INT_MAX;
        vector<vector<int>> ans;
        for (int i=0; i+1<a.size(); ++i) {
            int d = a[i+1] - a[i];
            if (d < dis) {
                dis = d;
                ans = vector<vector<int>>{{a[i], a[i+1]}};
            } else if (d == dis) 
                ans.push_back({a[i], a[i+1]});
        }
        return ans;
    }
};

File: 1200. Minimum Absolute Difference.py
class Solution:
    def minimumAbsDifference(self, a: List[int]) -> List[List[int]]:
        a.sort()
        mn = min(b-a for a,b in zip(a, a[1:]))
        return [[a,b] for a,b in zip(a, a[1:]) if b-a==mn]
            